src/Components/test/E2ETest/Tests/FormsTest.cs

```diff
@@ -432,6 +432,41 @@ public void InputComponentsRespondToAsynchronouslyAddedMessages()
             Browser.Equal("invalid", () => input.GetAttribute("class"));
         }
 
+        [Fact]
+        public void SelectComponentSupportsOptionsComponent() {
+            var appElement = Browser.MountTestComponent<SelectVariantsComponent>();
+            var input = appElement.FindElement(By.Id("input-value"));
+            var showAdditionalOptionButton = appElement.FindElement(By.Id("show-additional-option"));
+            var selectWithComponent = appElement.FindElement(By.Id("select-with-component"));
+            var selectWithoutComponent = appElement.FindElement(By.Id("select-without-component"));
+
+            // Select with custom options component and HTML component behave the
+            // same when the selectElement.value is provided
+            Browser.Equal("B", () => selectWithoutComponent.GetAttribute("value"));
+            Browser.Equal("B", () => selectWithComponent.GetAttribute("value"));
+
+            // Reset to a value that doesn't exist
+            input.Clear();
+            input.SendKeys("D\t");
+
+            // Confirm that both values are cleared
+            Browser.Equal("", () => selectWithComponent.GetAttribute("value"));
+            Browser.Equal("", () => selectWithoutComponent.GetAttribute("value"));
+
+            // Dynamically showing the fourth option updates the selected value
+            showAdditionalOptionButton.Click();
```

Comment:
This is great. I was going to ask for this case to be covered, but you're already there!